### PR TITLE
Bug 1895099: Fix VSphere UPI not populating PlatformStatus

### PR DESCRIPTION
**Pull request by bcrochet**
The VSphere UPI install does not fully populate PlatformStatus. When we deduped
the templates, we didn't account for this being nil. This should render these
templates empty when that PlatformStatus is nil.

**- What I did**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by patrickdillon** [comment]
cc @jcpowermac

**Comment by patrickdillon** [comment]
Not sure if these are setup:
/test e2e-vsphere
/test e2e-vsphere-upi

File: templates/common/vsphere/files/etc-systemd-system-crio-stream-address.conf.yaml
Comment (patrickdillon) [inline templates/common/vsphere/files/etc-systemd-system-crio-stream-address.conf.yaml:9]: don't we need to replace this with the simpler conditional used in other templates? `{{ if (onPremPlatformAPIServerInternalIP .) -}}`

**Comment by patrickdillon** [comment]
https://github.com/openshift/machine-config-operator/blob/master/templates/common/on-prem/files/NetworkManager-mdns-hostname.yaml also needs to be checked for `onPremPlatformAPIServerInternalIP`

**Comment by patrickdillon** [comment]
as does: 
https://github.com/openshift/machine-config-operator/blob/master/templates/common/on-prem/files/NetworkManager-onprem.conf.yaml

https://github.com/openshift/machine-config-operator/blob/master/templates/common/on-prem/files/mdns-publisher.yaml

https://github.com/openshift/machine-config-operator/blob/master/templates/common/on-prem/units/nodeip-configuration.service.yaml

File: templates/common/vsphere/files/etc-systemd-system-crio-stream-address.conf.yaml
Comment (bcrochet) [reply templates/common/vsphere/files/etc-systemd-system-crio-stream-address.conf.yaml:9]: There is no reference to the PlatformStatus in that file. Are you saying you don't want it at all?

**Comment by patrickdillon** [comment]
CI is consistently failing with `Failed pulling builder image` not sure what's going on there. I do wonder if the issue fixed by this pr could be causing part of the problem, as 4.7 images are not being promoted.

/lgtm

**Comment by jcpowermac** [comment]
MCO is still configured for template vs. multistep at least for UPI.

**Comment by bcrochet** [comment]
This may not pass until #2209 is merged. It might need a rebase after.

**Comment by jcpowermac** [comment]
@bcrochet looking at job results the installation succeeded the tests failed. I think this could move forward as-is.

@patrickdillon wdyt?

**Comment by bcrochet** [comment]
> @bcrochet looking at job results the installation succeeded the tests failed. I think this could move forward as-is.
> 
> @patrickdillon wdyt?

If you're ok with the order, it doesn't matter to me. This patch would probably allow #2209 to pass.

**Review by yuqi-zhang — APPROVED** [review]

**Comment by patrickdillon** [comment]
> > @bcrochet looking at job results the installation succeeded the tests failed. I think this could move forward as-is.
> > @patrickdillon wdyt?
> 
> If you're ok with the order, it doesn't matter to me. This patch would probably allow #2209 to pass, or vice versa. This one is probably failing because of the problem fixed in the #2209.

All vSphere installs fail without this PR. Merging this PR will move us closer to stability in CI and I don't think we should wait for a clean CI run to merge. I'm fine waiting a bit if it's better to merge #2209 first but I'd rather get the periodic back on track and use that as a barometer.

**Comment by bcrochet** [comment]
e2e-gcp-op seems to be failing across the board.

**Comment by jcpowermac** [comment]
Any chance we can get an override for gcp-op? This PR is needed for vSphere UPI to install and CI to pass.

**Comment by yuqi-zhang** [comment]
Ok given the AWS still slowly recovering, and the fact that this has passed all the other tests in the past + it should not affect cloud platforms, I will override this.

/override ci/prow/e2e-gcp-op

**Pull request merged**
